susi_python/main.py

```diff
@@ -3,221 +3,221 @@
+
+    def answer_from_json(self, json_response):
+        response_dict = json.loads(json_response)
+        query_response = get_query_response(response_dict)
+        return self._generate_result(query_response)
+
+
+    def _get_rss_entities(self, data):
+        entities = []
+        for datum in data:
+            values = datum.values
+            entity = RssEntity(
+                title=values['title'],
+                description=values['description'],
+                link=values['link']
+            )
+            entities.append(entity)
+        return entities
+
+    def add_device(self, access_token, room_name):
+
+        get_device_info = self.api_endpoint + '/aaa/listUserSettings.json?'
+        add_device_url = self.api_endpoint + '/aaa/addNewDevice.json?'
+        mac = get_mac()
+        macid = ':'.join(("%012X"%mac)[i:i+2] for i in range(0,12,2))
+
+        param1 = {
+            'access_token':access_token
         }
 
-        for dev in device:
-            if dev == macid:
-                print('Device already configured')
-                return
-            else :
-                adding_device = requests.post(add_device_url, params2)
-                print(adding_device.url)
-
-def sign_in(email, password, room_name=None):
-    global access_token
-    params = {
-        'login': email,
-        'password': password
-    }
-    sign_in_url = api_endpoint + '/aaa/login.json?type=access-token'
-    api_response = requests.get(sign_in_url, params)
-
-    if api_response.status_code == 200:
-        response_dict = api_response.json()
-        parsed_response = get_sign_in_response(response_dict)
-        access_token = parsed_response.access_token
         # print(access_token)
+
         if access_token is not None:
-            add_device(access_token, room_name)
-    else:
-        access_token = None
-
-def sign_up(email, password):
-    params = {
-        'signup': email,
-        'password': password
-    }
-    sign_up_url = api_endpoint + '/aaa/signup.json'
-    api_response = requests.get(sign_up_url, params)
-    parsed_dict = api_response.json()
-    return get_sign_up_response(parsed_dict)
-
-
-def forgot_password(email):
-    params = {
-        'forgotemail': email
-    }
-    forgot_password_url = api_endpoint + '/aaa/recoverpassword.json'
-    api_response = requests.get(forgot_password_url, params)
-    parsed_dict = api_response.json()
-    return get_forgot_password_response(parsed_dict)
-
-
-def get_previous_responses():
-    memory_url = api_endpoint + '/susi/memory.json'
-    api_response = requests.get(memory_url)
-    parsed_dict = api_response.json()
-    return get_memory_responses(parsed_dict)
+            device_info_response = requests.get(get_device_info,param1)
+            device_info = device_info_response.json()
+
+        # print(device_info)
+
+        if device_info is not None:
+            device = device_info['devices'] # list of existing mac ids
+            print(device)
```

Comment:
Still `print`.

Reply:
@hongquan , I can't see any logger logs integrated in this repo. I'll still check it out.